src/libAtomVM/term.h

```diff
@@ -740,25 +793,27 @@ static inline int term_binary_data_size_in_terms(uint32_t size)
  */
 static inline term term_from_literal_binary(const void *data, uint32_t size, Context *ctx)
 {
-    int size_in_terms = term_binary_data_size_in_terms(size);
+    if (size < REFC_BINARY_MIN) {
+        int size_in_terms = term_binary_data_size_in_terms(size);
+
+        term *boxed_value = memory_heap_alloc(ctx, size_in_terms + 1);
+        boxed_value[0] = (size_in_terms << 6) | 0x24; // heap binary
+        boxed_value[1] = size;
 
-    term *boxed_value = memory_heap_alloc(ctx, size_in_terms + 1);
-    boxed_value[0] = (size_in_terms << 6) | 0x24; // heap binary
-    boxed_value[1] = size;
+        if (data != NULL) {
```

Comment:
this special case might be completely avoid if we don't use this function for implementing `term_create_uninitialized_binary`, If so we might use `term_create_uninitialized_binary` as first step in this function.

Reply:
See previous comment.  Done.